lib/opencensus/span_context.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Opencensus.SpanContext do
+  @moduledoc """
+  Elixir convenience translation of `:opencensus.span_ctx`.
+
+  Most likely to be of use while writing unit tests, or packages that deal with spans.
+  Less likely to be of use while writing application code.
+  """
+
+  require Record
+  @fields Record.extract(:span_ctx, from_lib: "opencensus/include/opencensus.hrl")
+  Record.defrecordp(:span_ctx, @fields)
+
+  defstruct Keyword.keys(@fields)
+
+  @doc """
+  Convert a span context.
+
+      iex> :opencensus.span_ctx()
+      :undefined
+      iex> :opencensus.span_ctx() |> Opencensus.SpanContext.from()
+      nil
+
+      iex> trace_id = 158162877550332985110351567058860353513
+      iex> span_id = 13736401818514315360
+      iex> span_ctx = {:span_ctx, trace_id, span_id, 1, :undefined}
+      iex> Opencensus.SpanContext.from(span_ctx)
+      %Opencensus.SpanContext{
+        span_id: 13736401818514315360,
+        trace_id: 158162877550332985110351567058860353513,
+        trace_options: 1,
+        tracestate: :undefined
+      }
+  """
+  @spec from(:opencensus.span_ctx() | :undefined) :: %__MODULE__{}
+  def from(record)
```

Comment:
What does defining an empty function mean in elixir?

Reply:
It lets you attach the `@spec` and provide default arguments once, which then apply to all the other clauses for that function.

Reply:
@tsloughter it is function declaration like in C.